### PR TITLE
Fix compiler warnings and squash a buffer overflow

### PR DESCRIPTION
**Pull request by itkovian**
- Several C-lib or system calls are not checked for their return value
- Fixed a potential buffer overflow
- Initialise variables

### CODE REVIEW

File: src/resmom/mom_job_func.c [hunk 1]
```diff
@@ -613,15 +614,29 @@ int job_unlink_file(
   if ((setuid_ext(pjob->ji_qs.ji_un.ji_momt.ji_exuid, TRUE) == -1))
     {
     saved_errno = errno;
-    setegid(gid);
+   rc = setegid(gid);
+   if (!rc) {
```
Comment (wpoely86) [inline src/resmom/mom_job_func.c]: `rc` is -1 on error and 0 on success

Comment (wpoely86) [inline src/resmom/mom_job_func.c]: also indentation is wrong.

File: src/resmom/mom_job_func.c [hunk 2]
```diff
@@ -613,15 +614,29 @@ int job_unlink_file(
...
+      snprintf(log_buffer, sizeof(log_buffer),
+           "setegid(%lu) failed, error: %s\n",
+                    (unsigned long)pbsgroup,
```
Comment (wpoely86) [inline src/resmom/mom_job_func.c]: gid?

File: src/resmom/mom_job_func.c [hunk 3]
```diff
@@ -613,15 +614,29 @@ int job_unlink_file(
...
   setuid_ext(uid, TRUE);
-  setegid(gid);
+   rc = setegid(gid);
+   if (!rc) {
+      snprintf(log_buffer, sizeof(log_buffer),
+           "setegid(%lu) failed, error: %s\n",
+                    (unsigned long)pbsgroup,
+                    strerror(errno));
+      log_err(errno, __func__, log_buffer);
+    }
```
Comment (wpoely86) [inline src/resmom/mom_job_func.c]: same as above

File: src/resmom/mom_job_func.c [hunk 4]
```diff
@@ -685,7 +700,15 @@ void remove_tmpdir_file(
           }
         
         setuid_ext(pbsuser, TRUE);
-        setegid(pbsgroup);
+   int rc = setegid(pbsgroup);
+   if (!rc) {
```
Comment (wpoely86) [inline src/resmom/mom_job_func.c]: same

File: src/resmom/prolog.c [hunk 5]
```diff
@@ -1146,7 +1152,14 @@ void prepare_and_run_pelog_as_child(
       (which == PE_EPILOGUSERJOB))
     {
     setuid_ext(pbsuser, TRUE);
-    setegid(pbsgroup);
+    int rc = setegid(pbsgroup);
+    if (!rc) {
```
Comment (wpoely86) [inline src/resmom/prolog.c]: same

**Comment by itkovian** [comment]
Because I would prefer things to actually just work.

**Comment by dbeer** [comment]
What error do you get when a fail_unless() call doesn't have 3 arguments? Passing an empty string seems odd.

**Pull request closed without merging**